test/ipns-pubsub.js

```diff
@@ -107,64 +108,43 @@ const subscribeToReceiveByPubsub = (nodeA, nodeB, idA, idB, callback) => {
   const topic = `${namespace}${base64url.encode(keys.routingKey.toBuffer())}`
 
   // try to resolve a unpublished record (will subscribe it)
-  nodeB.api.name.resolve(idA, (err) => {
+  try {
+    await nodeB.api.name.resolve(idA)
+  } catch (err) {
     expect(err).to.exist() // not found
 
-    series([
-      (cb) => waitForPeerToSubscribe(nodeB.api, topic, cb),
-      (cb) => nodeB.api.pubsub.subscribe(topic, checkMessage, cb),
-      (cb) => waitForNotificationOfSubscription(nodeA.api, topic, idB, cb),
-      (cb) => nodeA.api.name.publish(ipfsRef, { resolve: false }, cb),
-      (cb) => waitFor(() => subscribed === true, (50 * 1000), cb),
-      (cb) => nodeB.api.name.resolve(idA, cb)
-    ], (err, res) => {
-      expect(err).to.not.exist()
-      expect(res).to.exist()
-
-      expect(res[3].name).to.equal(idA) // Published to Node A ID
-      expect(res[5]).to.equal(ipfsRef)
-
-      callback()
-    })
-  })
+    await waitForPeerToSubscribe(nodeB.api, topic)
+    await nodeB.api.pubsub.subscribe(topic, checkMessage)
+    await waitForNotificationOfSubscription(nodeA.api, topic, idB)
+    const res1 = await nodeA.api.name.publish(ipfsRef, { resolve: false })
+    await waitFor(() => subscribed === true, (50 * 1000))
+    const res2 = await nodeB.api.name.resolve(idA)
+
+    expect(res1.name).to.equal(idA) // Published to Node A ID
+    expect(res2).to.equal(ipfsRef)
+  }
```

Comment:
So, if `await nodeB.api.name.resolve(idA)` happens to run correctly, instead of failing, this test will pass. Please make a return by the end of catch + `thrown Error` after the catch

Reply:
See my comment about `expect(..).to.be.rejected()` above for how to not need to throw anything.

Reply:
What about `expect.fail(...)`?

Reply:
I did change it to `expect(..).to.be.rejected()` as @achingbrain suggested.